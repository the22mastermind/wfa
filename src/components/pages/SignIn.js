import React, { useState } from 'react';
import { Redirect } from 'react-router-dom';

const SignIn = () => {
  const [isAuthenticated, setIsAuthenticated] = useState(false);

  // Fake Authentication
  const handleSignIn = async (e) => {
    e.preventDefault();
    setIsAuthenticated(true);
  }

  if (isAuthenticated) {
    return (
      <Redirect to={{ pathname: '/dashboard' }} />
    );
  }
  return (
    <div className="container">
      <form className="form-wrapper" onSubmit={handleSignIn}>
        <div className="form-title">Sign In</div>
        <div className="input-wrapper">
          <input type="email" required placeholder="Email" className="input" />
        </div>
        <div className="input-wrapper">
          <input type="password" required placeholder="Password" className="input" />
        </div>
        <div className="submit-btn-wrapper">
          <input type="submit" className="submit-btn" value="Sign in" />
        </div>
      </form>
    </div>
  );
}

export default SignIn;
