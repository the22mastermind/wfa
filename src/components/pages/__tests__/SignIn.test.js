import React from 'react';
import renderer from 'react-test-renderer';

import SignIn from '../SignIn';


it('matches the Sign In snapshot', () => {
  const tree = renderer.create(<SignIn/>).toJSON();
  expect(tree).toMatchSnapshot();
});
