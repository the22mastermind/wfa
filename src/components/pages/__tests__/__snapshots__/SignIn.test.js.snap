// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the Sign In snapshot 1`] = `
<div
  className="container"
>
  <form
    className="form-wrapper"
    onSubmit={[Function]}
  >
    <div
      className="form-title"
    >
      Sign In
    </div>
    <div
      className="input-wrapper"
    >
      <input
        className="input"
        placeholder="Email"
        required={true}
        type="email"
      />
    </div>
    <div
      className="input-wrapper"
    >
      <input
        className="input"
        placeholder="Password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="submit-btn-wrapper"
    >
      <input
        className="submit-btn"
        type="submit"
        value="Sign in"
      />
    </div>
  </form>
</div>
`;
